stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -106,20 +133,25 @@ struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
 
       // Create nested autodiff copies of all shared arguments that do not point
       //   back to main autodiff stack
+      /*
       auto args_tuple_local_copy = apply(
           [&](auto&&... args) {
             return std::tuple<decltype(deep_copy_vars(args))...>(
                 deep_copy_vars(args)...);
           },
           args_tuple_);
+      */
+
+      scoped_args_tuple& args_tuple_scope = local_args_tuple_.local();
+      auto& args_tuple_local = *(args_tuple_scope.args_tuple_holder_);
 
       // Perform calculation
       var sub_sum_v = apply(
           [&](auto&&... args) {
             return ReduceFunction()(local_sub_slice, r.begin(), r.end() - 1,
                                     msgs_, args...);
           },
-          args_tuple_local_copy);
+          args_tuple_local);
 
       // Compute Jacobian
       sub_sum_v.grad();
```

Comment:
[side Q]

When we call grad here on a thread local `ChainableStack`, is it going to iterate through all of the vari that were created in that thread instance? Would it be faster if we had a `grad` function like

```cpp
template <typename Vari>
void grad(size_t end_pos, size_t begin_pos) {
  ChainableStack::instance_->var_stack_[end_pos,]->init_dependent();
  for (i = end_pos; i > begin_pos; --i) {
     ChainableStack::instance_->var_stack_[i]->chain();
  }
}
```

Then when we start the nested we save the starting place, then get the last place of the var_stack here and call

```cpp
grad(ChainableStack::instance_->var_stack.size() - 1, begin_nested.begin_pos());
```

Reply:
We only iterate here through the nested stuff, not more...but these must be traversed. So I think we can’t save her anything. Maybe I misunderstand your thoughts though.

Reply:
Oh whups yeah my bad I forgot `grad()` checks for whether it's in a nested already